### ISSUE TITLE
New-AzureRmSqlDatabase -Tags parameter does not work

### ISSUE THREAD

**Issue by treierts** [Azure/azure-powershell#1799]
I've tried to use the **-Tags** parameter with _New-AzureRmSqlDatabase_ but it does not seem to store the tags in Azure.

Also, the type required for **-Tags** is `System.Collections.Generic.Dictionary[string,string]` which differs from **-Tags** used in for example _New-AzureRmResourceGroup_ which is `System.Collections.ArrayList`.

The same type difference also exists with _New-AzureRmSqlServer_.

This is how I've created my $tags:

```
    $tags = New-Object "System.Collections.Generic.Dictionary[string,string]"
    $tags.Add("Company", "Company name")
    $tags.Add("Service", "Service name")
```

No error is returned, but the tags are not stored in Azure.

**Comment by markcowl** [comment]
SQL Team:  The guidance is to use Hashtable for tags - also, you should create the tags if they do not exist.

@treierts thanks for the report.  Have you tried creating the tag first using New-AzureRmTag and then using the tag in your cmdlets?

**Comment by jamestao** [comment]
Fixed Tags overall in #3206.
